artifact-size-enforcer/src/test/java/org/codice/ddf/admin/beta/enforcer/ArtifactSizeEnforcerRuleTest.java

```diff
@@ -0,0 +1,216 @@
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.beta.enforcer;
+
+import static org.codice.ddf.admin.beta.enforcer.ArtifactSizeEnforcerRule.PROJECT_ARTIFACT_ID_PROP;
+import static org.codice.ddf.admin.beta.enforcer.ArtifactSizeEnforcerRule.PROJECT_BUILD_DIR_PROP;
+import static org.codice.ddf.admin.beta.enforcer.ArtifactSizeEnforcerRule.PROJECT_PACKAGING_PROP;
+import static org.codice.ddf.admin.beta.enforcer.ArtifactSizeEnforcerRule.PROJECT_VERSION_PROP;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.fail;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugin.logging.SystemStreamLog;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+import org.codehaus.plexus.component.repository.exception.ComponentLookupException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ArtifactSizeEnforcerRuleTest {
+
+    private static final String SAMPLE_ARTIFACT_ID = "SampleArtifactId";
+    private static final String SAMPLE_VERSION = "SampleVersion";
+    private static final String SAMPLE_ARTIFACT_FILE_NAME = SAMPLE_ARTIFACT_ID + "-" + SAMPLE_VERSION + ".jar";
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+    private File sampleBuildDir;
+    private MockEnforcementRuleHelper defaultMockhelper;
+
+    @Before
+    public void before() throws IOException {
+        sampleBuildDir = tempFolder.getRoot();
+        defaultMockhelper = new MockEnforcementRuleHelper().artifactId(SAMPLE_ARTIFACT_ID)
+                .packaging(ArtifactSizeEnforcerRule.JAR)
+                .version(SAMPLE_VERSION)
+                .buildDir(sampleBuildDir.getPath());
+    }
+    @Test
+    public void unknownPackaging() {
+        ArtifactSizeEnforcerRule enforcer = new ArtifactSizeEnforcerRule();
+        defaultMockhelper.packaging("UnknownPackageType");
+        try {
+            enforcer.execute(defaultMockhelper);
```

Comment:
I'm okay with this, but this test will pass as long as there is no exception thrown. So if someone modifies this in the future, this would pass the method returns for any reason, not just an unknown package type. For instance, if skip was set to true, this test would still pass, but not for the correct reason. Might be safer to add an assertion that the `getPackaging()` method was called once.